### ISSUE TITLE
java.lang.IllegalArgumentException: Trying to execute a sync operation with a storage manager for an anonymous account

### ISSUE THREAD

**Issue by semmakara6666** [nextcloud/android#13679]
### Cause of error
```java
Exception in thread "Thread-5" java.lang.IllegalArgumentException: Trying to execute a sync operation with a storage manager for an anonymous account
    at com.owncloud.android.operations.common.SyncOperation.execute(SyncOperation.java:49)
    at com.owncloud.android.ui.activity.DrawerActivity.lambda$fetchExternalLinks$9(DrawerActivity.java:1231)
    at com.owncloud.android.ui.activity.DrawerActivity.$r8$lambda$oHM2j1MZkviAIdyIFWoI2kSeF9c(Unknown Source:0)
    at com.owncloud.android.ui.activity.DrawerActivity$$ExternalSyntheticLambda6.run(D8$$SyntheticClass:0)
    at java.lang.Thread.run(Thread.java:1012)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30290290`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `a02q`
* Model: `SM-A025F`
* Id: `SP1A.210812.016`
* Product: `a02qnnxx`

### Firmware
* SDK: `31`
* Release: `12`
* Incremental: `A025FXXS8CXH1`

**Comment by semmakara6666** [comment]
> ### Cause of error
> ```java
> Exception in thread "Thread-5" java.lang.IllegalArgumentException: Trying to execute a sync operation with a storage manager for an anonymous account
>     at com.owncloud.android.operations.common.SyncOperation.execute(SyncOperation.java:49)
>     at com.owncloud.android.ui.activity.DrawerActivity.lambda$fetchExternalLinks$9(DrawerActivity.java:1231)
>     at com.owncloud.android.ui.activity.DrawerActivity.$r8$lambda$oHM2j1MZkviAIdyIFWoI2kSeF9c(Unknown Source:0)
>     at com.owncloud.android.ui.activity.DrawerActivity$$ExternalSyntheticLambda6.run(D8$$SyntheticClass:0)
>     at java.lang.Thread.run(Thread.java:1012)
> 
> ```
> 
> ### App information
> * ID: `com.nextcloud.client`
> * Version: `30290290`
> * Build flavor: `gplay`
> 
> ### Device information
> * Brand: `samsung`
> * Device: `a02q`
> * Model: `SM-A025F`
> * Id: `SP1A.210812.016`
> * Product: `a02qnnxx`
> 
> ### Firmware
> * SDK: `31`
> * Release: `12`
> * Incremental: `A025FXXS8CXH1`

**Comment by joshtrichards** [comment]
Duplicate of #13323 
Fixed in #13440
